File: code/stable/projectile/projectile_c_p_nol_b_u_v_d/src/java/Projectile/Control.java
package Projectile;

/** \file Control.java
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Controls the flow of the program
*/
import java.io.FileNotFoundException;
import java.io.IOException;

public class Control {
    
    /** \brief Controls the flow of the program
        \param args List of command-line arguments
    */
    public static void main(String[] args) throws FileNotFoundException, IOException {
        String filename = args[0];
        double g = 9.8;
        double epsilon = 2.0e-2;
        InputParameters inParams = new InputParameters(filename);
        double t_flight = Calculations.func_t_flight(inParams, g);
        double p_land = Calculations.func_p_land(inParams, g);
        double d_offset = Calculations.func_d_offset(inParams, p_land);
        String s = Calculations.func_s(inParams, epsilon, d_offset);
        OutputFormat.write_output(s, d_offset, t_flight);
    }
}

File: People/Sam/TestGen/Committee Meeting 1 - June 2023/assets/code/Control.py
## \file Control.py
# \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
# \brief Controls the flow of the program
import sys
import argparse

from . import Calculations
from . import InputParameters
from . import OutputFormat

def main():	
	# from https://stackoverflow.com/questions/54071312/how-to-pass-command-line-argument-from-pytest-to-code
	parser = argparse.ArgumentParser(description='Control')
	parser.add_argument('file')
	args = parser.parse_args()
	filename = f"{args.file}"

	# filename = sys.argv[1]

	g = 9.8
	epsilon = 2.0e-2
	inParams = InputParameters.InputParameters(filename)
	t_flight = Calculations.func_t_flight(inParams, g)
	p_land = Calculations.func_p_land(inParams, g)
	d_offset = Calculations.func_d_offset(inParams, p_land)
	s = Calculations.func_s(inParams, epsilon, d_offset)
	OutputFormat.write_output(s, d_offset, t_flight)

if __name__ == "__main__":
	main()
